maintainers/testing.md

```diff
@@ -87,6 +87,28 @@ Overall goal, follow the open source ecosystem for infrastructure choices
     - I.E. cuDF PyTests:
       - [https://github.com/rapidsai/cudf/tree/branch-0.7/python/cudf/tests](https://github.com/rapidsai/cudf/tree/branch-0.7/python/cudf/tests)
 
+## Datasets
+
+- Many tests depend on the presence of specific datasets in order to properly verify code correctness.
+- Some datasets are small enough that they can be included with the source code, but others are relatively large and cannot practically coexist with the source. In both cases, the test executable should not assume datasets are located in a particular place (either with a relative path, since the executable could be moved, or an absolute path, since different machines will likely use different absolute paths), so the location of the datasets must be provided at test runtime.
+- There are three mechanisms that allow tests to locate datasets on disk:
```

Comment:
The descriptions of each mechanism are long. Why not state something like: "Tests can locate datasets on disk using three methods which are tried in order until one returns non-null" Then remove the lines in each about when it triggers.

Reply:
I rewrote how the mechanisms are used in the new document and I think it's clearer now.